selfdrive/car/ford/fordcan.py

```diff
@@ -4,161 +4,164 @@
-
-  Ford lane centering command uses a third order polynomial to describe the road centerline. The
-  polynomial is defined by the following coefficients:
-    c0: lateral offset between the vehicle and the centerline
-    c1: heading angle between the vehicle and the centerline
-    c2: curvature of the centerline
-    c3: rate of change of curvature of the centerline
-  As the PSCM combines this information with other sensor data, such as the vehicle's yaw rate and
-  speed, the steering angle cannot be easily controlled.
-
-  The PSCM should be configured to accept TJA/LCA commands before these commands will be processed.
-  This can be done using tools such as Forscan.
-
-  Frequency is 20Hz.
-  """
-
-  values = {
-    "LatCtlRng_L_Max": 0,                                   # Unknown [0|126] meter
-    "HandsOffCnfm_B_Rq": 0,                                 # Unknown: 0=Inactive, 1=Active [0|1]
-    "LatCtl_D_Rq": lca_rq,                                  # Mode: 0=None, 1=ContinuousPathFollowing, 2=InterventionLeft, 3=InterventionRight, 4-7=NotUsed [0|7]
-    "LatCtlRampType_D_Rq": ramp_type,                       # Ramp speed: 0=Slow, 1=Medium, 2=Fast, 3=Immediate [0|3]
-    "LatCtlPrecision_D_Rq": precision,                      # Precision: 0=Comfortable, 1=Precise, 2/3=NotUsed [0|3]
-    "LatCtlPathOffst_L_Actl": path_offset,                  # Path offset [-5.12|5.11] meter
-    "LatCtlPath_An_Actl": path_angle,                       # Path angle [-0.5|0.5235] radians
-    "LatCtlCurv_NoRate_Actl": curvature_rate,               # Curvature rate [-0.001024|0.00102375] 1/meter^2
-    "LatCtlCurv_No_Actl": curvature,                        # Curvature [-0.02|0.02094] 1/meter
-  }
-  return packer.make_can_msg("LateralMotionControl", CANBUS.main, values)
-
-
-def create_lkas_ui_command(packer, main_on: bool, enabled: bool, steer_alert: bool, hud_control, stock_values: dict):
-  """
-  Creates a CAN message for the Ford IPC IPMA/LKAS status.
-
-  Show the LKAS status with the "driver assist" lines in the IPC.
-
-  Stock functionality is maintained by passing through unmodified signals.
-
-  Frequency is 1Hz.
-  """
-
-  # LaActvStats_D_Dsply
-  #    R  Intvn Warn Supprs Avail No
-  # L
-  # Intvn  24    19    14     9   4
-  # Warn   23    18    13     8   3
-  # Supprs 22    17    12     7   2
-  # Avail  21    16    11     6   1
-  # No     20    15    10     5   0
-  #
-  # TODO: test suppress state
-  if enabled:
-    lines = 0  # NoLeft_NoRight
-    if hud_control.leftLaneDepart:
-      lines += 4
-    elif hud_control.leftLaneVisible:
-      lines += 1
-    if hud_control.rightLaneDepart:
-      lines += 20
-    elif hud_control.rightLaneVisible:
-      lines += 5
-  elif main_on:
-    lines = 0
-  else:
-    if hud_control.leftLaneDepart:
-      lines = 3  # WarnLeft_NoRight
-    elif hud_control.rightLaneDepart:
-      lines = 15  # NoLeft_WarnRight
+class FordCAN:
```

Comment:
any reason this is a class?

Reply:
just to remove packer arg, and then when we get around to CAN-FD we will need to specify a different bus too

Reply:
i'd rather not introduce new patterns here

Reply:
oh I just saw that tesla already did it https://github.com/commaai/openpilot/blob/master/selfdrive/car/tesla/teslacan.py

Reply:
I can revert for now